pyfr/plugins/tavg.py

```diff
@@ -357,3 +362,248 @@ def __call__(self, intg):
+            anames = [fields[i].removeprefix('avg-') for i in aidx]
+            amap = [i for n, i in sorted(zip(anames, aidx))]
+
+            sidx = [i for i, v in enumerate(fields) if
+                    re.match(r'\bstd-(?!fun-avg-)', v)]
+            snames = [fields[i].removeprefix('std-') for i in sidx]
+            smap = [i for n, i in sorted(zip(snames, sidx))]
+
+            if i == 0:
+                self.stats, self.cfg = stats, cfg
+
+                prec = cfg.get('backend', 'precision')
+                self.dtype = np.float32() if prec == 'single' else np.float64()
+
+                self.fields = fields
+                self.cfgsect = cfgsect
+                self.region = cfg.get(cfgsect, 'region')
+                self.uuid = file['mesh_uuid']
+
+                self.idxs = [k for k in file if re.search(r'_idxs_p\d+$', k)]
+                self.dkeys = [k for k in file if
+                              re.match(r'tavg_[^\W_]+_p\d+$', k)]
+
+                self.mapping = [(amap, smap)]
+                fnames = [x.removeprefix('fun-avg-') for x in fields if
+                          x.startswith('fun-avg-')]
+                self.anames = sorted(anames)
+                self.fnames = sorted(fnames)
+
+                # Initialise std max and avg registers
+                if self.std_all:
+                    self.std_max = np.zeros(len(anames))
+                    self.std_avg = np.zeros(len(anames))
+                    self.fstd_max = np.zeros(len(fnames))
+                    self.fstd_avg = np.zeros(len(fnames))
+
+                # Build fun avg expressions
+                c = cfg.items_as('constants', float)
+                self.fexprs = [cfg.getexpr(cfgsect, f'fun-avg-{k}', subs=c)
+                               for k in fnames]
+            else:
+                self.mapping.append((amap, smap))
+
+            # Check for compatibility of files
+            if self.uuid != file['mesh_uuid']:
+                raise RuntimeError('Average files computed on different meshes')
+            if self.region != cfg.get(cfgsect, 'region'):
+                raise RuntimeError('Average files computed on different regions')
+
+            for k in cfg.items(cfgsect, prefix='avg-'):
+                if cfg.get(cfgsect, k) != self.cfg.get(self.cfgsect, k):
+                    raise RuntimeError('Different average field definitions')
+
+    def _merge_data(self, outf):
+        file0, stats, cfg, dt = self.files[0]
+        amap0, smap0 = self.mapping[0]
+
+        for key in self.dkeys:
+            # Initialise accumulators
+            data = file0[key]
+            avg_acc = np.zeros_like(data[:, amap0])
+            var_acc = np.zeros_like(data[:, smap0])
+
+            # Perform the rest of the accumulations
+            t = 0
+            for files, mapping in zip(self.files, self.mapping):
+                file, stats, cfg, dt = files
+                amap, smap = mapping
```

Comment:
Can avoid these by saying `adata, sdata = data[:, mapping]` or something close to that

Reply:
The files and mapping a created in different places, I think this is the cleanest way to do it.